File: base_java/src/main/java/com/hesheng1024/base_java/network/NetService.java
package com.hesheng1024.base_java.network;

import com.hesheng1024.base_java.network.rx_util.ResponseFunc;
import com.hesheng1024.base_java.network.rx_util.SchedulerTransformer;

import io.reactivex.rxjava3.core.Observable;

/**
 * 具体网络请求类
 *
 * @author hesheng1024
 * @date 2019/5/22 16:08
 */
public class NetService {
    private final IApiService mIApiService;

    NetService(IApiService apiService) {
        this.mIApiService = apiService;
    }

    public IApiService getApiService() {
        return mIApiService;
    }

    public Observable<String> demo(String name) {
        return mIApiService.demo(name)
                .compose(SchedulerTransformer.transformer())
                .map(new ResponseFunc<>());
    }
}

File: base_kotlin/src/main/java/com/hesheng1024/base_kotlin/network/NetService.kt
package com.hesheng1024.base_kotlin.network

/**
 *
 * @author hesheng1024
 * @date 2020/6/18 10:41
 */
class NetService(apiService: IApiService) {
    private val mIApiService: IApiService = apiService

    fun getApiService(): IApiService {
        return mIApiService
    }

}